### ISSUE TITLE
Fix TypeError for media queries in indentation

### ISSUE THREAD

**Issue by ryprice** [stylelint/stylelint#5003]
> Which rule, if any, is the bug related to?

indentation

> What code is needed to reproduce the bug?

```css
.fancyButton
  color: red

  @media only screen and (min-width: 600px)
    color: blue
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "indentation": 2,
  }
}
```

> Which version of stylelint are you using?

13.7.2
This worked previously in 9.9.0, and I verified that with this isolated example.

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`npx stylelint src/fancyButton.sass --syntax sass`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

sass

> What did you expect to happen?

I'm using stylelint inside of webpack and this exception was really hard to debug because the only thing webpack was telling me was `ERROR in Cannot read property 'type' of undefined` without indicating the file or plugin that this exception was coming from. This is not the root issue, but it would have been nice for the stylelint webpack plugin to somehow identify itself as the thrower of the exception, as well as the file it failed to parse.

Anyway, I was able to isolate the issue to the described command line setup, and I would have expected no errors.

> What actually happened (e.g. what warnings or errors did you get)?

From npx stylelint: (line number might be slightly off because I was debugging with console.logs. I believe the error [comes from this line](https://github.com/stylelint/stylelint/blob/master/lib/rules/indentation/index.js#L149).)
```
TypeError: Cannot read property 'type' of undefined
    at indentationLevel (D:\src\listlab-web\node_modules\stylelint\lib\rules\indentation\index.js:157:20)
    at D:\src\listlab-web\node_modules\stylelint\lib\rules\indentation\index.js:62:22
    at D:\src\listlab-web\node_modules\postcss\lib\container.js:135:18
    at Rule.each (D:\src\listlab-web\node_modules\postcss\lib\container.js:101:16)
    at Rule.walk (D:\src\listlab-web\node_modules\postcss\lib\container.js:131:17)
    at D:\src\listlab-web\node_modules\postcss\lib\container.js:148:24
    at Root.each (D:\src\listlab-web\node_modules\postcss\lib\container.js:101:16)
    at Root.walk (D:\src\listlab-web\node_modules\postcss\lib\container.js:131:17)
    at D:\src\listlab-web\node_modules\stylelint\lib\rules\indentation\index.js:56:8
    at D:\src\listlab-web\node_modules\stylelint\lib\lintPostcssResult.js:118:8
```

From webpack (where I was originally running this):
`ERROR in Cannot read property 'type' of undefined`

**Comment by rovadi** [comment]
Hey @ryprice this issue it's related with this [4865](https://github.com/stylelint/stylelint/issues/4865), I posted a solution there that works for this same case, hope it helps.

**Comment by ryprice** [comment]
@rovadi do you think it would be worth wrapping rule execution in a try catch to avoid complete failure? I made a PR to implement this.

**Comment by jeddy3** [comment]
Thanks for the report and for using the template.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

> From npx stylelint: (line number might be slightly off because I was debugging with console.logs. I believe the error comes from this line.)

I think you're right. We may just need to add a guard before accessing`node.parent.type`, or there maybe something bigger amiss.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
